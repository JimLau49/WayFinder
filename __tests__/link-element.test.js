import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import LinkElement from "../src/components/building-information/link-element.component";

const mockLinkItem = {
  id: 1,
  title: "Engineering and Computer Science Association",
  link: "https://ecaconcordia.ca/",
};

describe("LinkItem component", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(<LinkElement linkItem={mockLinkItem} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should check if link has proper title", () => {
    const wrapper = shallow(<LinkElement linkItem={mockLinkItem} />);

    const link = wrapper.find({
      testID: "Engineering and Computer Science Association",
    });

    expect(link.props().children).toEqual(mockLinkItem.title);
  });

  it("should check if clicking department links is functional", () => {
    const wrapper = shallow(<LinkElement linkItem={mockLinkItem} />);
    const mockOpenURL = jest.fn();
    jest.mock("react-native/Libraries/Linking/Linking", () => {
      return {
        openURL: mockOpenURL,
      };
    });

    const link = wrapper.find({
      testID: "Engineering and Computer Science Association",
    });

    link.simulate("press");
    expect(mockOpenURL).toHaveBeenCalledTimes(1);
    expect(mockOpenURL).toBeCalledWith(mockLinkItem.link);
  });
});
