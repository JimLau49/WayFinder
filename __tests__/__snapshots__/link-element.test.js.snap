// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkItem component should match snapshot 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    â€¢  
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "blue",
        "fontSize": 16,
        "textDecorationLine": "underline",
      }
    }
    testID="Engineering and Computer Science Association"
  >
    Engineering and Computer Science Association
  </Text>
</View>
`;
